### ISSUE TITLE
[3.0] collection configuration admin.hidden is not working properly

### ISSUE THREAD

**Issue by mpresecan** [payloadcms/payload#6243]
### Describe the Bug

I.e: This will still show collection in admin:

```javascript
export const sessions: CollectionConfig = {
    slug: 'sessions',
    admin: {
        hidden: true
    },
...
}
```

Does not seems to be inlign with docs: [https://payloadcms.com/docs/configuration/collections#admin-options](https://payloadcms.com/docs/configuration/collections#admin-options)

### To Reproduce

Descriebed in the bug section

### Payload Version

3.0.0-beta.19

**Comment by PatrikKozak** [comment]
Hey @mpresecan - I just gave this a shot and was not able to reproduce - when adding the `admin.hidden` prop on a collection, it properly hides it from view in the admin UI for me.

Can you try again on the latest beta version and if you are still experiencing the issue, can you please attach a reproducible repo?

Thank you!

**Comment by mpresecan** [comment]
Hi @PatrikKozak, I just tried to launch a new instance of payload app, and I could not reproduce the bug either. Now, when I checked my original project, I saw that my collection dissapeared from the admin ui, as it is in the config file. But when I enabled it again, then did not show. I recon this is the issue due to some cache, but I tried to reproduce the same thing in my blank new payload app, but could make it work to test it out, I got some other issues.

I will close this issue thread, and if I figure out what might be causing this behaviour I will make a new issue, with detailed description. Thank you for your help.
